### PR TITLE
Feature/11890 cert revocation/12580 check certificate

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Adds the check for a certificate against the revocation list from rest service cache.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12580

### CODE REVIEW

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/Extensions/HealthCertifcateService+Inits.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import XCTest	
+import Foundation
+import HealthCertificateToolkit
+@testable import ENA
+
+extension HealthCertificateService {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/Extensions/HealthCertifcateService+Inits.swift:10]: File name has an i missing (HealthCertif**i**cateService)

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/Extensions/HealthCertifcateService+Inits.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	convenience init(
+		store: HealthCertificateStoring,
+		dccSignatureVerifier: DCCSignatureVerifying,
+		dscListProvider: DSCListProviding,
+		appConfiguration: AppConfigurationProviding,
+		digitalCovidCertificateAccess: DigitalCovidCertificateAccessProtocol = DigitalCovidCertificateAccess(),
+		notificationCenter: UserNotificationCenter = UNUserNotificationCenter.current(),
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/Extensions/HealthCertifcateService+Inits.swift:17]: I think we need to pass on these arguments, currently they are ignored

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateValidator.swift [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		)
+		let cachedResult = restServiceProvider.cached(resource)
+		
+		if case .success(let revocationChunk) = cachedResult {
+			let revocationChunkMatches = revocationChunk.hashes.contains { chunkHash in
+				chunkHash == hash.dataWithHexString()
+			}
+			if revocationChunkMatches {
+				return true
+			}
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateValidator.swift:70]:
```suggestion
			return revocationChunkMatches
```

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/Extensions/HealthCertifcateService+Inits.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/Extensions/HealthCertifcateService+Inits.swift:17]: Good point, thanks!

**Review by nickguendling — APPROVED** [review]

**Pull request merged**